crowbar_framework/app/models/ceph_service.rb

```diff
@@ -166,9 +166,9 @@ def create_proposal
 
     # Any spare node after allocating mons and osds is fair game
     # to automatically use as the calamari server
-    calamari_nodes = select_nodes_for_role(nodes, "ceph-calamari")
+    calamari_nodes = select_nodes_for_role(nodes - osd_nodes - mon_nodes, "ceph-calamari")
     calamari_nodes.reject! do |n|
```

Comment:
> Doesn't this suffer from exactly the same problem as the mds node selection

You are right (although with calamari it is not such problem when it is not deployed at all)

> reject all cluster members before calling 

That's a good point